internal/dispatch/graph/reachableresources_test.go

```diff
@@ -603,3 +609,109 @@ func TestCaveatedReachableResources(t *testing.T) {
 		})
 	}
 }
+
+func TestReachableResourcesWithConsistencyLimitOf1(t *testing.T) {
+	defer goleak.VerifyNone(t, goleakIgnores...)
+
+	ctx, dispatcher, revision := newLocalDispatcherWithConsistencyLimit(t, 1)
```

Comment:
can we somehow assert that no more than one goroutine is spawn? There are already tests in `TaskRunner`, but would be nice to assert the dispatcher properly integrates it.

Reply:
I'm not sure how we could reasonably do so. Any ideas?

Reply:
I have no clue honestly, was hoping there is some library out there that helps with that

Reply:
https://pkg.go.dev/runtime#NumGoroutine exists but we'd have to check exactly when it is running, which seems... problematic